addons/account/models/account_move_line.py

```diff
@@ -632,6 +632,8 @@ def get_rate(from_currency, to_currency, company, date):
                 date=date,
             )
         for line in self:
+            if not line.currency_id:
+                line._compute_currency_id()
```

Comment:
```suggestion
            if not line.currency_id:
                line.currency_rate = 1
```
According to @tsb-odoo, we prefer not to do a smart thing behind the user.
As the `currency_id` is required, the user can't save without it. To avoid the traceback, you can set the `currency_rate` to 1.